core/dbt/task/deps.py

```diff
@@ -287,6 +298,13 @@ def _checkout(self, project):
 
     def _fetch_metadata(self, project):
         path = self._checkout(project)
+        if self.version[0] == 'master' and self.warn_unpinned:
+            dbt.exceptions.warn_or_error(
```

Comment:
this is in the class of warnings that's worth breaking out `log_fmt=printer.yellow('{}')` for :)

Reply:
`log_fmt=printer.red_bold_blinking('{}')`

Reply:
Also, can we format this as:
```
WARNING: The git package "https://github.com/fishtown-analytics/dbt-utils" is not pinned.
         This can introduce breaking changes into your project without warning!

See https://docs.getdbt.com/docs/package-management#section-specifying-package-versions
```

We show this warning if a revision is unspecified too, so calling out `master` here might be confusing. I think we're best off pointing people to the docs, then doing a really good job of explaining what's wrong and how to proceed in there.

As a part of this, I'll clean up the whole `/package-management` docs page for sure